gensim/models/word2vec.py

```diff
@@ -1819,18 +1820,31 @@ def predict_output_word(self, context_words_list, topn=10):
         if not hasattr(self.wv, 'vectors') or not hasattr(self, 'syn1neg'):
             raise RuntimeError("Parameters required for predicting the output words not found.")
 
-        word2_indices = [self.wv.get_index(w) for w in context_words_list if w in self.wv]
-        if not word2_indices:
-            logger.warning("All the input context words are out-of-vocabulary for the current model.")
-            return None
+        if all(isinstance(w, int) for w in context_words_list):
+            # then, indices were passed. Check they are valid
```

Comment:
```suggestion
            # Integer indices were passed. Check that they are all valid.
```

Reply:
This is my first pull request ever, and even after reading the docs, I am not sure what to do now. It is also about doing things the usual way in this project. So, given the github interface, it looks like I have to validate your changes and commit them. Do you want me to commit things little by little, or batch-wise? Do you want a detailed commit message here or is github's default one enough?

Reply:
Radim (@piskvorky) as project owner can offer a more authoritative answer, but I'd suggest:

1. If you find a suggestion helpful/noncontroversial, the main thing is to get it into your working branch (that's feeding into this PR, specifically `M-Demay:vectorize-predict_output_word`) in whatever way is most convenient for you - that could be via this web UI, but might just be via your own local editing, Generally, commits in fewer batches of related/contemporaneous changes that you're sure all work together are good. (This would especially be the case for a bunch of related code-comment touch-ups: they'd all fit into a nice batch called something like "improved comment language".) But at the same time, if it's helpful or important  to do a micro-fix right away with a tiny commit, by all means get it in rather than let it wait for other work. Generally, if you wind up fixing the relevant line in some other way - or making it irrelevant – this PR web UI will note the comments/suggestion as obsolete.
2. If the suggestion is not an obvious improvement, by all means discuss or counter-suggest. 
3. I would *not* want Github's web ui generic default commit-message (here `Update gensim/models/word2vec.py`) - it's filler that adds nothing. But even a super-simple actually descriptive commit-comment, "better comment", would be true & useful.
4. I think other preferred suggestions below are actually likely to make this particular code-comment obsolete/irrelevant, so consider & integrate suggestions as a whole, not one-by-one, so ensure the overall best steps are taken.